models/comments.model.js

```diff
@@ -0,0 +1,57 @@
+const db = require('../db/connection');
+const { checkArticleIdExists } = require("../models/articles.model.js")
+
+exports.fetchCommentsById = (article_id) => {
+  
+
+
+    const fetchCommentsPromise = (article_id) =>{
+        
+        const query = `
+            SELECT * FROM comments
+            WHERE article_id = $1
+            ORDER BY created_at DESC;`
+    
+        return db.query(query, [article_id]).then((result) => {
+            if (result.rows.length === 0) {
+                return Promise.reject({status:200, msg: "article_id has no comments"})
+            } else {
+                return result.rows;
+            }
+        });
+    }
+
+
+    //checks article_id exists && fetches comments
+    return Promise.all([fetchArticleById(article_id), fetchCommentsPromise(article_id)])
```

Comment:
It makes more sense for the Promise.all to come first so we can reject any requests where the article does not exist first. If I try to query article 99 in the comments model, I get nothing back, which would run the Promise.reject on line 17, even though the reason it should reject is because the article does not exist

Reply:
Do you mean put the fetchCommentsPromise first in the promise all?

Reply:
The Promise.all should happen in the controller so that if one of these requests fail they both fail.